pydantic_settings/sources.py

```diff
@@ -877,6 +892,49 @@ def _read_file(self, file_path: Path) -> dict[str, Any]:
             return yaml.safe_load(yaml_file)
 
 
+class AzureKeyVaultSettingsSource(PydanticBaseSettingsSource):
+    _secret_client: SecretClient  # type: ignore
+
+    def __init__(
+        self,
+        settings_cls: type[BaseSettings],
+        url: str,
+        credential: TokenCredential,  # type: ignore
+    ) -> None:
+        import_azure_key_vault()
+        super().__init__(settings_cls)
+        self._secret_client = SecretClient(vault_url=url, credential=credential)  # type: ignore
+
+    def get_field_value(self, field: FieldInfo, field_name: str) -> tuple[Any, str, bool]:
+        field_value: Any | None = None
+
+        # It's not possible to use underscores in Azure Key Vault
+        secret_name = field_name.replace('_', '-')
```

Comment:
I think you need to use [_extract_field_info](https://github.com/pydantic/pydantic-settings/blob/09d10094d97425853829c2971051a561ccd4a2d2/pydantic_settings/sources.py#L199) to respect the field alias if defined.

Reply:
This is complex and I don't see the point in aliasing secret names. Can we iterate in the next version?

Reply:
This is the basic functionality that we already have in all of the source classes.

You need to move the `_extract_field_info` to `PydanticBaseSettingsSource` class and then call it here. it returns a list of tuples that contain `field_key`, `env_name` and `value_is_complex`. you can use the `env_name` to fetch the value from key vault . there are some use-cases of `_extract_field_info` in the code, you can check them as well.

Reply:
This is the basic functionality that we already have in all of the source classes.
Why is this complex? you need to call the

Reply:
I've added the alias support.